cmake/templates/python_install.sh.in

```diff
@@ -22,12 +22,11 @@ echo_and_run mkdir -p "$DESTDIR@CMAKE_INSTALL_PREFIX@/@PYTHON_INSTALL_DIR@"
 # into one location when some dependencies were installed in another
 # location, #123.
 echo_and_run /usr/bin/env \
+    @PIP_EXTRA_ENV@ \
+    PIP_BREAK_SYSTEM_PACKAGES=1 \
     PYTHONPATH="@CMAKE_INSTALL_PREFIX@/@PYTHON_INSTALL_DIR@:@CMAKE_BINARY_DIR@/@PYTHON_INSTALL_DIR@:$PYTHONPATH" \
-    CATKIN_BINARY_DIR="@CMAKE_BINARY_DIR@" \
-    "@PYTHON_EXECUTABLE@" \
-    "@CMAKE_CURRENT_SOURCE_DIR@/setup.py" \
-    @SETUPTOOLS_EGG_INFO@ \
-    build --build-base "@CMAKE_CURRENT_BINARY_DIR@" \
-    install \
-    --root="${DESTDIR-/}" \
-    @SETUPTOOLS_ARG_EXTRA@ --prefix="@CMAKE_INSTALL_PREFIX@" --install-scripts="@CMAKE_INSTALL_PREFIX@/@CATKIN_GLOBAL_BIN_DESTINATION@"
+    "@PYTHON_EXECUTABLE@" -m pip install --prefix "@CMAKE_INSTALL_PREFIX@" --ignore-installed --no-deps --use-pep517 --no-build-isolation \
+    "@CMAKE_CURRENT_SOURCE_DIR@"
```

Comment:
This creates empty `build/` folders in each project dir.
Here is a workaround to have pip forward the `build` command to setup.py with an option to use a different build directory in the workspace build folder
(also needed in the bat if we go for that road)

```suggestion
    "@PYTHON_EXECUTABLE@" -m pip install \
      --prefix "@CMAKE_INSTALL_PREFIX@" \
      --ignore-installed \
      --no-deps \
      --use-pep517 \
      --no-build-isolation \
      --config-settings=--global-option=build \
      --config-settings=--global-option=--build-base=@CMAKE_BINARY_DIR@/setuptools_build \
      "@CMAKE_CURRENT_SOURCE_DIR@"
```

Also, I tested devel workspaces for a bit and the previous logic there remains working.